native_toolchain_rust_common/lib/src/manifest.dart

```diff
@@ -0,0 +1,118 @@
+import 'dart:io';
+
+import 'package:logging/logging.dart';
+import 'package:package_config/package_config.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart';
+
+class NdkManifestInfo {
+  NdkManifestInfo({
```

Comment:
I am a bit torn on this. Yes, both NdkManifestInfo and RustManifestInfo only have one field, but that's not intentional, it just happened to be that way. Maybe in future new fields are added, which means it will get reverted to a normal class, which to me indicates that maybe, conceptually, it's not really an extension type.